components/rule-mgt/org.wso2.carbon.identity.rule.management/src/main/java/org/wso2/carbon/identity/rule/management/util/RuleBuilder.java

```diff
@@ -154,37 +155,58 @@ private void addORCombinedRule() {
         orRuleCount++;
     }
 
-    private void validateExpression(Expression expression) {
+    private Expression validateExpression(Expression expression) {
```

Comment:
Shall we rename the this method, as this build and return the express in addition to the validation.

Reply:
Make sense. What about 'validateAndResolveExpression'

Reply:
Will go with 'validateExpressionAndResolveValue'

Reply:
Fixed